File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/ui/AlertDialogHandler.java
package com.emerchantpay.gateway.genesisandroid.api.ui;

import android.app.AlertDialog;
import android.content.Context;

public class AlertDialogHandler {

    private AlertDialog.Builder dialogBuilder;
    private AlertDialog alert;

    public AlertDialogHandler(Context context, String title, String message) {
        super();

        dialogBuilder = new AlertDialog.Builder(context);
        dialogBuilder.setTitle(title);
        dialogBuilder.setMessage(message);
        dialogBuilder.setNeutralButton("OK", null);

        alert = dialogBuilder.create();
    }

    public void show() {
        alert.show();
    }

    public void dismiss() {
        alert.dismiss();
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/ui/AlertDialogHandler.kt
package com.emerchantpay.gateway.genesisandroid.api.ui

import android.app.AlertDialog
import android.content.Context


open class AlertDialogHandler(context: Context, title: String, message: String) {

    private var dialogBuilder: AlertDialog.Builder = AlertDialog.Builder(context)
    private var alert: AlertDialog? = null

    init {
        dialogBuilder.setTitle(title)
        dialogBuilder.setMessage(message)
        dialogBuilder.setNeutralButton("OK", null)

        alert = dialogBuilder.create()
    }

    fun show() {
        alert?.show()
    }

    fun dismiss() {
        alert?.dismiss()
    }
}